File: workspace/amr_ws/src/amr_py_examples/amr_py_examples/simple_listener.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class SimpleListener(Node):

    def __init__(self):
        super().__init__("simple_listener")
        self.sub_ = self.create_listener(String, "chatter", self.msgCallback, 10)
        self.sub_

    def msgCallback(self, msg):
        self.get_logger().info("I heard: %s" % msg.data)


def main():
    rclpy.init()

    simple_listener = SimpleListener()
    rclpy.spin(simple_listener)
    
    simple_listener.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: workspace/amr_ws/src/amr_cpp_examples/src/simple_listener.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>


using std::placeholders::_1;

class SimpleListener : public rclcpp::Node
{
public:
  SimpleListener() : Node("simple_listener")
  {
    sub_ = create_listener<std_msgs::msg::String>(
        "chatter", 10, std::bind(&SimpleSubscriber::msgCallback, this, _1));
  }

private:
  rclcpp::Listener<std_msgs::msg::String>::SharedPtr sub_;

  void msgCallback(const std_msgs::msg::String &msg) const
  {
    RCLCPP_INFO_STREAM(this->get_logger(), "I heard: " << msg.data.c_str());
  }
};


int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<SimpleListener>());
  rclcpp::shutdown();
  return 0;
}